plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -932,7 +932,14 @@ public function search_orders( $term ) {
 		 * @param int[]  $order_ids Search results as an array of order IDs.
 		 * @param string $term      The search term.
 		 */
-		return array_map( 'intval', (array) apply_filters( 'woocommerce_cot_shop_order_search_results', $order_ids, $term ) );
+		$order_ids = array_map( 'intval', (array) apply_filters( 'woocommerce_cot_shop_order_search_results', $order_ids, $term ) );
+
+		/**
+		 * Adding actual compatibility as well with CPT's woocommerce_shop_order_search_results.
+		 *
+		 * @since 7.8.0.
+		 */
+		return array_map( 'intval', (array) apply_filters( 'woocommerce_shop_order_search_results', $order_ids, $term ) );
```

Comment:
The original hook supplies three params (`$order_ids, $term, $search_fields`) whereas here we only have the first two. I wonder if this could be confusing/frustrating for 3PDs. 

What do you think of moving these hooks to `OrdersTableQuery`? If we also add a getter (for the private `search_fields` property) to `OrdersTableSearchQuery` then we could probably provide all three params.

Reply:
Good catch!  Seems like doing this is a little tricky because search_orders is wrapper over the query in a way, where the output of `OrdersTableQuery` is not compatible with the output for this filter. Specifically, output for this filter only modifies the IDs that are returned, whereas, the query output also returns total count, and also may returns fields other then IDs, entire object etc.

What do you think about calling another instance of search fields here, alternatively, I am also open to refactoring `wc_order_search` to accepts more params that it can pass to the search order methods, and then applying these two filters in the `wc_order_search` method. wdyt?

Reply:
@vedanshujain: first, so sorry for the delay in responding. I did receive a notification, but accidentally cleared it 😬 

> Seems like doing this is a little tricky because search_orders is wrapper over the query in a way, where the output of OrdersTableQuery is not compatible with the output for this filter.

Yep, could definitely be good reasons not to. Fwiw, this is (broadly) how I picture execution flowing at present, when HPOS is authoritative:

```
         wc_order_search()
                 ↓
OrdersTableDataStore::search_orders()
                 ↓
          wc_get_orders()
                 ↓
     WC_Order_Query::get_orders()
                 ↓
    OrdersTableDataStore::query()  →  OrdersTableQuery::__construct()
                 |                                  ↓
                 |                     OrdersTableQuery::run_query()
                 ↓                     -----------------------------
              (returns
               orders)
```

Inside [`OrdersTableQuery::run_query()`](https://github.com/woocommerce/woocommerce/blob/92a48d1934bf9e5940b6d00ee28f7131d0da7685/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php#L1130) we are working with an array of order IDs (this is before the result array is formed, which actually happens here in `OrdersTableDataStore`) so, within the `run_query` method, it occurs we could do something like:

```php
		// Run query.
		$this->orders = array_map( 'absint', $wpdb->get_col( $this->sql ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared

		// Maybe apply search filter.
		if ( null !== $this->search_query ) {
			array_map( 
				'intval', 
				(array) apply_filters( 
					'woocommerce_shop_order_search_results', 
					$this->orders, 
					$this->search_query->get_search_term(),
					$this->search_query->get_search_fields(),
				) 
			);
		}
```

It would however also mean implementing those final setter methods.

> What do you think about calling another instance of search fields here, alternatively, I am also open to refactoring wc_order_search to accepts more params that it can pass to the search order methods, and then applying these two filters in the wc_order_search method.

These could also be fine: I think the main thing is to ensure compatibility with the existing hook, if we can reasonably do so.